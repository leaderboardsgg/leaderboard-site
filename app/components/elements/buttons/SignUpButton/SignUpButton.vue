<script setup lang="ts">
import BaseButton from 'elements/buttons/BaseButton/BaseButton.vue'
</script>

<template>
  <BaseButton class="sign-up-button" v-bind="$attrs">
    {{ $t('signup') }}
  </BaseButton>
</template>

<style lang="postcss" scoped>
.sign-up-button {
  @apply bg-gray-100 hover:bg-black mx-1 py-1 text-opacity-70 hover:text-white;
}
</style>
