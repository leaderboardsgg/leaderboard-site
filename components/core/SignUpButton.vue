<template>
  <BaseButtonLink
    class="sign-up-button"
    to="#"
    v-bind="$attrs"
    v-on="$listeners"
  >
    Sign Up
  </BaseButtonLink>
</template>

<style scoped>
  .sign-up-button {
    @apply bg-gray-100 hover:bg-black mx-1 py-1 text-opacity-70 hover:text-white;
  }
</style>
