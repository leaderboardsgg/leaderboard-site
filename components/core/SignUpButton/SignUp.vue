<template>
  <BaseButtonLink
    :class-names="classNames + `
      bg-gray-100 text-opacity-70
      py-2
      hover:bg-black hover:text-white
    `"
    link-to="#"
  >
    Sign Up
  </BaseButtonLink>
</template>

<script>
export default {
  props: {
    classNames: {
      default: '',
      type: String
    }
  }
};
</script>
