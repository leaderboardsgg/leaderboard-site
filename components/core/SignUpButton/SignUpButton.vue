<template>
  <BaseButton class="sign-up-button" v-bind="$attrs" v-on="$listeners">
    Sign Up
  </BaseButton>
</template>

<style scoped>
.sign-up-button {
  @apply bg-gray-100 hover:bg-black mx-1 py-1 text-opacity-70 hover:text-white;
}
</style>
