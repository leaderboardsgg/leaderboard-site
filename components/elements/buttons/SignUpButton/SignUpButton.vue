<template>
  <BaseButton class="sign-up-button" v-bind="$attrs" v-on="$listeners">
    Sign Up
  </BaseButton>
</template>

<script lang="ts">
import { defineComponent } from '@nuxtjs/composition-api';
import BaseButton from '@/components/elements/buttons/BaseButton.vue';

export default defineComponent({
  name: 'SignUpButton',
  components: {
    BaseButton,
  },
});
</script>

<style lang="postcss" scoped>
.sign-up-button {
  @apply bg-gray-100 hover:bg-black mx-1 py-1 text-opacity-70 hover:text-white;
}
</style>
