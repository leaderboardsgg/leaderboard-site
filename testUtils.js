import Vue from 'vue';
import { RouterLinkStub } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (Component, options) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
