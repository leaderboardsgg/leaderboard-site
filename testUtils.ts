import Vue, { ComponentOptions } from 'vue';
import { RouterLinkStub, VueClass } from '@vue/test-utils';
import userEvent from '@testing-library/user-event';
import { render, RenderOptions } from '@testing-library/vue';

export const NuxtLinkStub = Vue.component<
  { tag: string; to: string },
  unknown,
  unknown
>('NuxtLinkStub', {
  extends: RouterLinkStub,
  render(h) {
    const temp = h(
      this.tag || 'a',
      {
        attrs: {
          ...this.$vnode.data?.attrs,
          ...this.$attrs,
          href: this.to,
        },
      },
      this.$slots.default,
    );

    return temp;
  },
});

export const stubbedRender = (
  Component: VueClass<Vue> | ComponentOptions<Vue>,
  options?: RenderOptions<Vue>,
) => {
  return render(Component, {
    stubs: { NuxtLink: NuxtLinkStub, ...options?.stubs },
    ...options,
  });
};

export * from '@testing-library/vue';
export * from '@vue/test-utils';
export { userEvent as fireEvent };
