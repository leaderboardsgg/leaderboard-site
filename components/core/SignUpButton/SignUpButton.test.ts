import { stubbedRender } from '@/testUtils';
import SignUpButton from './SignUpButton.vue';

describe('<SignUpButton />', () => {
  it('should render without crashing', () => {
    const { unmount } = stubbedRender(SignUpButton);

    unmount();
  });

  it('renders correctly', () => {
    const { container } = stubbedRender(SignUpButton);

    expect(container.firstChild).toMatchSnapshot();
  });
});
