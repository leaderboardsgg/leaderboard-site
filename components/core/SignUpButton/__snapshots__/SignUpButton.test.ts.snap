// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpButton /> renders correctly 1`] = `
<a
  class="button-link sign-up-button"
  href="#"
  to="#"
>
  
  Sign Up

</a>
`;
